pkg/providers/pricing/pricing.go

```diff
@@ -451,3 +393,17 @@ func populateInitialSpotPricing(pricing map[string]float64) map[string]zonal {
 	}
 	return m
 }
+
+func (p *Provider) Reset() {
```

Comment:
Can we reuse this to populate the pricing initially so that we don't duplicate the code here?